huaweicloud/services/iotda/resource_huaweicloud_iotda_device_policy.go

```diff
@@ -0,0 +1,297 @@
+	)
+
+	client, err := cfg.NewServiceClientWithDerivedAuth("iotda", region, isDerived)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA client: %s", err)
+	}
+
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildCreateDevicePolicyBodyParams(d)),
+	}
+
+	createResp, err := client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA device policy: %s", err)
+	}
+
+	createRespBody, err := utils.FlattenResponse(createResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	policyId := utils.PathSearch("policy_id", createRespBody, "").(string)
+	if policyId == "" {
+		return diag.Errorf("error creating IoTDA device policy: ID is not found in API response")
+	}
+
+	d.SetId(policyId)
+
+	return resourceDevicePolicyRead(ctx, d, meta)
+}
+
+func resourceDevicePolicyRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg       = meta.(*config.Config)
+		region    = cfg.GetRegion(d)
+		isDerived = WithDerivedAuth(cfg, region)
+		httpUrl   = "v5/iot/{project_id}/device-policies/{policy_id}"
+	)
+
+	client, err := cfg.NewServiceClientWithDerivedAuth("iotda", region, isDerived)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA client: %s", err)
+	}
+
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = strings.ReplaceAll(getPath, "{policy_id}", d.Id())
+	getOpts := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	getResp, err := client.Request("GET", getPath, &getOpts)
+	if err != nil {
+		// When the resource does not exist, query API will return `404` error code.
+		return common.CheckDeletedDiag(d, err, "error retrieving IoTDA device policy")
+	}
+
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("policy_name", getRespBody, nil)),
+		d.Set("statement", flattenStatement(utils.PathSearch("statement", getRespBody, nil))),
+		d.Set("space_id", utils.PathSearch("app_id", getRespBody, nil)),
+		d.Set("created_at", utils.PathSearch("create_time", getRespBody, nil)),
+		d.Set("updated_at", utils.PathSearch("update_time", getRespBody, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenStatement(statementResp interface{}) []map[string]interface{} {
+	if statementResp == nil {
+		return nil
+	}
+
+	statementList := statementResp.([]interface{})
+	if len(statementList) < 1 {
+		return nil
+	}
+
+	result := make([]map[string]interface{}, 0)
+	for _, v := range statementList {
+		statementMap := v.(map[string]interface{})
+		statement := map[string]interface{}{
+			"effect":    statementMap["effect"],
```

Comment:
```suggestion
			"effect":    utils.PathSearch("effect", v, nil),
```

It is unnecessary to convert v to statementMap.

Reply:
done.